### PR TITLE
Centralize calls to library()

### LINKED ISSUE #91: Replace calls to library() with source("packages.R")
#88 introduced packages.R. We can now replace calls to `library()` with `source("packages.R")`.
This will remove duplication and thus leave the system cleaner.

It would be nice to write a helper that updates packages.R. The ui may be like 
`spelling::update_wordslist()` or `usethis::use_tidy_description()`. Maybe it should
live in r2dii.usethis.

### PR DESCRIPTION
**Pull request by maurolepore**
Closes #91 
Relates to #137 

This PR centralizes all calls to `library()` in a single place. This removes
duplication, makes the code base smaller, and makes it easier to prune
packages because now it is clearer which packages we depend on.

### CODE REVIEW

**Comment by cjyetman** [comment]
What is the `dry` directory for? Why is it called that?

I think maybe a more descriptive filename than `packages.R` might be helpful.

Otherwise, I support this plan and I don’t see any obvious errors.

File: web_tool_script_1.R [hunk 1]
```diff
@@ -5,19 +5,7 @@ options(encoding = "UTF-8")
 # devtools::install_github("2DegreesInvesting/r2dii")
 # install.packages("fst")
```
Comment (cjyetman) [inline web_tool_script_1.R:6]: should these commented out install lines also be removed from here and/or added to `dry/packages.R`?

File: 0_fund_processing.R [hunk 2]
```diff
@@ -1,5 +1,4 @@
-library(tidyverse)
```
Comment (jacobvjk) [inline 0_fund_processing.R:1]: @Clare2D this file is not needed for anything, right?

**Review by jacobvjk — COMMENTED** [review]
LGTM
from my side, as long as CJ's question can be answered

File: web_tool_script_1.R (hunk 1 above)
Comment (maurolepore) [reply web_tool_script_1.R:5]: I'll submit a separate PR to remove dead code like this one.

Comment (maurolepore) [reply web_tool_script_1.R:5]: https://github.com/2DegreesInvesting/PACTA_analysis/issues/144

File: .Rbuildignore [hunk 3]
```diff
@@ -43,4 +43,4 @@
 ^data-raw$
 ^LICENSE\.md$
 ^docker\.sh$
-^dry$
+^deduplicate$
```
Comment (maurolepore) [inline .Rbuildignore:46]: DRY is short for Don't Repeat Yourself (https://en.wikipedia.org/wiki/Don%27t_repeat_yourself). 
I removed the jargon, favoring a more explicit name -- thanks CJ.

File: 0_fund_processing.R (hunk 2 above)
Comment (maurolepore) [reply 0_fund_processing.R:1]: If so, maybe flag this in #144

File: R/utils.R [hunk 4]
```diff
@@ -1,12 +1,18 @@
...
+  path <- fs::path(
+    "deduplicate",
+    "set_portfolio-name-ref-all_working-location_and_web-parameters.R"
+  )
+  source(path)
+}
+
+use_r_packages <- function(path = "deduplicate/load-and-attach-r-packages.R") {
+  source(path)
 }
```
Comment (maurolepore) [inline R/utils.R:18]: This helper hides the details about where and how we load and attach packages -- which are a low level detail that is likely to change. This way the change will happen in just one place.

File: web_tool_script_2.R [hunk 5]
```diff
@@ -1,34 +1,22 @@
+options(r2dii_config = paste0(par_file_path,"/AnalysisParameters.yml"))
+
+devtools::load_all()
+use_r_packages()
```
Comment (maurolepore) [inline web_tool_script_2.R:4]: I reorganized scripts headers to start with options, then sourcing all functions and data, then attaching packages, then anything else.

**Review by maurolepore — COMMENTED** [review]
Here are some comments to highlight the most interesting changes.

File: .Rbuildignore (hunk 3 above)
Comment (cjyetman) [reply .Rbuildignore:46]: hahahaha… awesome, learn something new every day 😉

**Pull request merged**